seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/AbstractReadStrategy.java

```diff
@@ -89,34 +90,36 @@ boolean checkFileType(String path) {
 
     @Override
     public List<String> getFileNamesByPath(String path) throws IOException {
+        ArrayList<String> fileNames = new ArrayList<>();
+        Path basePath = new Path(path);
+        if (readPartitions.isEmpty()) {
+            fileNames.addAll(getFileNamesByPath(basePath));
+        } else {
+            for (String readPartition : readPartitions) {
+                fileNames.addAll(getFileNamesByPath(new Path(basePath, readPartition)));
+            }
```

Comment:
This way will missed some partition.
eg:
```
readPartition = ["age=11"]
filepath = basePath/location=newyork/age=11
```

Seem like our test case not enough. Could you add some test case to cover this case?

Reply:
I think if you have multi partitions, the partition name should be written in full. There is no 'age=11' partition. We can support wildcard matching, which can be written as 'location=*/age=11' or '*/age=11'. What do you think?"

Reply:
I mean, we supported the partition value like `["age=11"]` before, but after this change, it does not supported any more. It's unaccepted.

Reply:
Ok, I understand